src/main/java/TestFile2.java

```diff
@@ -0,0 +1,16 @@
+class TestFile2 {
+    public String foo() {
+        return "bar";
+    }
+
+    private static void incorrectAssignmentInIfCondition() {
+        boolean value = false;
+        if (value = true) {
+            //do Something
+            value = false;
+        }
+        if (value = false) {
```

Comment:
[FindBugs] WARNING: QBA: TestFile2.incorrectAssignmentInIfCondition() assigns boolean literal in boolean expression